### ISSUE TITLE
SnmpMessageExtension.IsRunningOnIOS property not reporting correctly when running Xamarin.iOS

### ISSUE THREAD

**Issue by jvanhuis-zebra** [lextm/sharpsnmplib#79]
IsRunningOnIOS property is reporting false while running on Xamarin.iOS.  It appears that the preprocessor definition of the XAMARINIOS10 property is missing.

From Messaging/SnmpMessageExtension.cs:
```
        /// <summary>
        /// Gets a value indicating whether it is running on iOS.
        /// </summary>
        /// <value><c>true</c> if is running on iOS; otherwise, <c>false</c>.</value>

        public static bool IsRunningOnIOS
        {
            get
            {
#if NET452
                return false;
#elif NETSTANDARD1_3
                return RuntimeInformation.IsOSPlatform(OSPlatform.OSX);
#elif XAMARINIOS10
                return true;
#else
                return false;
#endif
            }
        }
```

However, when running on iOS, the following exception is thrown (from Messaging/EndPointExtension.cs):

```
 Protocol option not supported
 
  at System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Int32 optionValue) [0x00030] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.12.0.4/src/Xamarin.iOS/mcs/class/System/System.Net.Sockets/Socket.cs:2484
  at System.Net.Sockets.Socket.SetSocketOption (System.Net.Sockets.SocketOptionLevel optionLevel, System.Net.Sockets.SocketOptionName optionName, System.Boolean optionValue) [0x00008] in /Library/Frameworks/Xamarin.iOS.framework/Versions/11.12.0.4/src/Xamarin.iOS/mcs/class/System/System.Net.Sockets/Socket.cs:2469
  at Lextm.SharpSnmpLib.Messaging.EndPointExtension.GetSocket (System.Net.EndPoint endpoint) [0x0002b] in <48c08cc23da9466abbce352035b39c9b>:0
  at Lextm.SharpSnmpLib.Messaging.SnmpMessageExtension+<GetResponseAsync>d__15.MoveNext () [0x0006f] in <48c08cc23da9466abbce352035b39c9b>:0
```

Further inspection of the library using Dotpeek dissassembly reveals this compiled implementation:
```
public static bool IsRunningOnIOS
    {
      get
      {
        return false;
      }
    }

```

**Comment by lextm** [comment]
Please review your solution structure, especially how MSBuild resolves the assemblies. If somehow MSBuild determines to use the .NET Standard 1.3 build, then of course it would return false.

The official sample project can be found here,

https://github.com/lextm/sharpsnmplib_xamarin/

**Comment by jvanhuis-zebra** [comment]
Not having any luck with that.  Actually, pulling down the sample project, I see the same exception being thrown when running on iOS.

**Comment by lextm** [comment]
Fixed the conditional compilation issue in 10.0.9 release.
